pkg/kratos/service.go

```diff
@@ -657,6 +696,47 @@ func (s *Service) HasTOTPAvailable(ctx context.Context, id string) (bool, error)
 	return ok, nil
 }
 
+func (s *Service) HasNotEnoughLookupSecretsLeft(ctx context.Context, id string) (bool, error) {
+
+	identity, _, err := s.kratosAdmin.IdentityApi().
+		GetIdentity(ctx, id).
+		IncludeCredential([]string{"lookup_secret"}).
+		Execute()
+
+	if err != nil {
+		return false, err
+	}
+
+	lookupCredentials := identity.GetCredentials()["lookup_secret"].Config["recovery_codes"]
+
+	jsonbody, err := json.Marshal(lookupCredentials)
+	if err != nil {
+		s.logger.Errorf("Marshalling to json failed: %s", err)
+		return false, err
+	}
+
+	lookupSecrets := new(LookupSecrets)
+	if err := json.Unmarshal(jsonbody, &lookupSecrets); err != nil {
+		s.logger.Errorf("Unmarshalling failed: %s", err)
+		return false, err
+	}
+
+	unusedCodes := 0
+	for _, code := range *lookupSecrets {
+		if code.UsedAt.IsZero() {
+			unusedCodes += 1
+		}
+	}
+
+	if unusedCodes > 3 {
+		return false, nil
+	}
```

Comment:
suggestion: this is great already, but I suggest you create at least a function to isolate this specific logic of determining if codes available are enough or not. I'm referring to lines 731-733